// babel-webpack/src/app/actions/index.js

import axios from 'axios';
import {expiredToken} from './auth';

import { UNAUTH_USER,
  FETCH_EMPRESAS,CREATE_EMPRESA, GET_EMPRESA } from './types'

const URL_EMPRESAS = "https://m00rjv766c.execute-api.us-east-1.amazonaws.com/dev/empresas/";

axios.defaults.baseURL = URL_EMPRESAS;
axios.defaults.headers.common['authorization'] = localStorage.getItem('token');

export function getEmpresa(nome) {
  const url = `${URL_EMPRESAS}${nome}`;
  return function(dispatch) {
    axios.get(url)
      .then(response => {
        const empresa = JSON.parse(response.data.empresa);
        console.log(empresa);
        dispatch({
          type: GET_EMPRESA,
          payload: empresa
        });
      })
      .catch( response => {
        if(expiredToken()){
          dispatch({
            type: UNAUTH_USER
          });
        }
      });
  }
}

export function fetchEmpresas() {
  return function(dispatch) {
    axios.get(URL_EMPRESAS)
      .then(response => {
        //console.log(JSON.parse(response.data.empresas));
        dispatch({
          type: FETCH_EMPRESAS,
          payload: JSON.parse(response.data.empresas)
        });
      })
      .catch( (error) => {
        if(expiredToken()){
          dispatch({
            type: UNAUTH_USER
          });
        }
      });
  }
}
export function createEmpresa(nome, descricao) {
  return function(dispatch) {
    axios.post(URL_EMPRESAS,{"nome": nome})
      .then(response => {
        console.log(JSON.parse(response.data));
        //console.log(response.data);
        dispatch({
          type: CREATE_EMPRESA,
          payload: nome
        });
      });
  }
}
