// babel-webpack/src/app/components/empresas/empresa.js
import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getEmpresa } from '../../actions/index';

class Empresa extends Component {

  componentWillMount(){
    this.props.getEmpresa(this.props.match.params.nome_empresa);
  }
  renderDescricao(){
    const descricao = this.props.empresas.selected.descricao;
    if(descricao === undefined || descricao === null || Object.keys(descricao).length === 0 ){
      return "Sem Descricao";
    } else {
      return "Descricao";
    }
  }

  render() {
    return (
      <Card>
        <CardTitle title={this.props.match.params.nome_empresa} subtitle="Cnae2" />
        <CardText>{this.renderDescricao()}
        </CardText>
        <CardActions>
          <FlatButton label="Riscos" />
          <FlatButton label="Trabalhadores" />
        </CardActions>
      </Card>
    )
  }
}


function mapStateToProps(state) {
  return { empresas: state.empresas}
}
function mapDispathToProps(dispatch) {
  return bindActionCreators({getEmpresa}, dispatch);
}

export default connect(mapStateToProps,mapDispathToProps)(Empresa);
