// babel-webpack/src/app/components/empresas/empresa_form.js
import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createEmpresa } from '../../actions/index';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

class EmpresaForm extends Component {

  constructor(props) {
    super(props);
    this.state = {
      nome: '',
      cnae: '',
      cnpj: ''
    };
  }

  handleNomeChange(e) {
    this.setState({nome: e.target.value});
  }

  handleSubmit(e) {
    e.preventDefault();
    const nome = this.state.nome.trim();
    const descricao = this.state.cnae.trim();
    this.props.createEmpresa(nome,descricao);
  }

  render() {
    return (
      <form>
      Criar Nova Empresa: <br />
      <TextField hintText="Nome"
      onChange={this.handleNomeChange.bind(this)}
      value={this.state.nome} /><br />
      <RaisedButton label="Salvar" primary={true}
      onClick={this.handleSubmit.bind(this)}/>
      </form>
    )
  }
}

function mapDispathToProps(dispatch) {
  return bindActionCreators({createEmpresa}, dispatch);
}
function mapStateToProps(state){
  return { auth: state.auth};
}
export default connect(null, mapDispathToProps )(EmpresaForm);
