// babel-webpack/src/app/components/header/search.js
import React, { Component } from 'react';
import IconMenu from 'material-ui/IconMenu';
import IconButton from 'material-ui/IconButton';
import MenuItem from 'material-ui/MenuItem';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import AutoComplete from 'material-ui/AutoComplete';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { signoutUser } from '../../actions/auth';


class SearchBar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      dataSource: []
    };
  }

  handleSignout(e){
    this.props.signoutUser();
  }
  handleUpdateInput = (value) => {
    this.setState({
      dataSource: [
        value,
        value + value,
        value + value + value,
      ],
    });
  };

  render() {
    return (
      <AutoComplete
      hintText="Type anything"
      dataSource={this.state.dataSource}
      onUpdateInput={this.handleUpdateInput}
      />
    )
  }
}
function mapDispathToProps(dispatch) {
  return bindActionCreators({signoutUser: signoutUser}, dispatch);
}
function mapStateToProps(state){
  return { authenticated: state.auth.authenticated};
}
export default connect(mapStateToProps, mapDispathToProps )(SearchBar);
